gslib/utils/copy_helper.py

```diff
@@ -1343,6 +1338,14 @@ def _ShouldDoParallelCompositeUpload(logger,
           file_size >= parallel_composite_upload_threshold):
     return False
 
+  # TODO(KMS, Compose): Until we ensure service-side that we have efficient
+  # compose functionality over objects with distinct KMS encryption keys (CMEKs)
+  # or distinct CSEKs, don't utilize parallel composite uploads.
+  if kms_keyname:
+      logger.warning('Not using parallel composite upload for KMS-encryption. ' +
+                     'This combination is not currently supported by GCS.')
+      return False
+
```

Comment:
Why wouldn't you just keep line 1300 in tact and add the warning up there?

I might also make line 1349-1389 a function, which you can call at 1300, so that way you don't have to duplicate this message. Something like:
```
if kms_keyname or _DoBucketMetdataPCUCheck():
  logger.warning('Parallel composite uploads have been disabled. Composite uploads are not ' +
                 'supported in buckets that have a default customer-managed encryption key.')
  return False
```

Reply:
I realized that I marked this as Resolved, but my comment was not posted.'

I talked to @mhandugan offline regarding this. The reason for doing this was to not log a warning if composite upload is not required (e.g when the files are small enough).